### ISSUE TITLE
Add tests

### ISSUE THREAD

**Issue by alexewerlof** [load1n9/openai#35]
Deno has native testing support. Could we add some tests with mock data just to ensure that the parsing and encoding/decoding functionality works as expected?

**Comment by lino-levan** [comment]
Tests would definitely be nice, especially around my SSE decoding.

**Comment by load1n9** [comment]
Yeah sounds good, a big thing for testing is testing is costly so keeping it to a minimum while testing everything would be nice

**Comment by alexewerlof** [comment]
To reduce duplication of effort, and diverging from whatever convention and preference you may have in mind, I'd prefer if you guys add the basic tests and I'll add to it with the PRs that I'll be making. 🙏
